pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1326,83 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// Retrieve configuration in form of Ironic's clean steps
+func (p *ironicProvisioner) buildCleanSteps(ironicNode *nodes.Node) (cleanSteps []nodes.CleanStep) {
+	// Check whether the clean steps for RAID configuration should be executed or not
+	if len(ironicNode.RAIDConfig) == 0 && p.host.Spec.RAID != nil {
```

Comment:
These criterial will leave any existing RAID configuration on a host that is brought into the cluster if the user tries to provision to it without setting their own RAID configuration. In an earlier conversation we talked about how the delete step always needs to be performed, even if there is no new RAID configuration to apply.

The logic for that might be simpler if buildRAIDCleanSteps() was not a separate method and the steps were added here in this method.

Reply:
Done. I made the `delete step` always be performed :)